src/db12/__main__.py

```diff
@@ -0,0 +1,163 @@
+--extra-iteration option  to avoid tail effects. In this case copies which
+finish early run additional iterations until all the measurements finish.
+The COPIES (ie an integer) argument causes multiple copies of the benchmark to
+be run in parallel. The tokens "wholenode", "jobslot" and "single" can be 
+given instead to use $MACHINEFEATURES/total_cpu, $JOBFEATURES/allocated_cpu, 
+or 1 as the number of copies respectively. If $MACHINEFEATURES/total_cpu is
+not available, then the number of (logical) processors visible to the 
+operating system is used.
+Unless the token "single" is used, the script prints the following results to
+two lines on stdout:
+COPIES SUM ARITHMETIC-MEAN GEOMETRIC-MEAN MEDIAN
+RAW-RESULTS
+The tokens "version" and "help" print information about the script.
+The source code of dirac_benchmark.py provides examples of how the functions
+within dirac_benchmark.py can be used by other Python programs.
+dirac_benchmark.py is distributed from  https://github.com/DIRACGrid/DB12
+"""
+
+    copies = None
+    iterations = 1
+    extra_iteration = False
+
+    parser = argparse.ArgumentParser()
+    #pylint: disable=line-too-long
+    parser.add_argument("--iterations", nargs='?', help="number of iterations to perform", default='')
+    parser.add_argument("--extra-iteration", nargs='?', help="whether an extra iteration is needed", default='')
+    parser.add_argument("--json", nargs='?', help="generate json files", default='')
+    parser.add_argument("copy", help="number of copies", nargs='?', const='', default='')
+    parser.add_argument('--version', action='version', version=VERSION, default='')
+
+    subparsers = parser.add_subparsers(dest='parser')
+    parser_single = subparsers.add_parser('single')
+    parser_single.set_defaults(func=single_dirac_benchmark_cli)
+
+    parser_wholenode = subparsers.add_parser('wholenode')
+    parser_wholenode.set_defaults(func=wholenode_dirac_benchmark_cli)
+
+    parser_jobslot = subparsers.add_parser('jobslot')
+    parser_jobslot.set_defaults(func=jobslot_dirac_benchmark_cli)
+
+    parser_multiple = subparsers.add_parser('multiple')
+    parser_multiple.set_defaults(func=multiple_dirac_benchmark_cli)
+
+    args = parser.parse_args()
+
+    if args.iterations:
+        iterations = int(args[13:])
+    elif args.extra_iteration:
+        extra_iteration = True
+    elif args.json:
+        try:
+            output = args.func(copies, extra_iteration, iterations)
+        except AttributeError:
+            output = single_dirac_benchmark_cli()
+        dump_as_json(args.json, output)
+    elif not args.copy.startswith("--"):
+        copies = args.copy
+    else:
+        parser = argparse.ArgumentParser(description=help_string)
+    try:
+        if copies in ('', "single"):
+            output = args.func()
+        else:
+            output = args.func(int(copies), extra_iteration, iterations)
+    except AttributeError:
+        output = single_dirac_benchmark_cli()
```

Comment:
I think that this can be removed, you can replace that with:
```suggestion
args.func(args)
```

About the `--json` flag: it will be automatically passed in parameter of the called function, then in the function you can check: if `json` is True, then I call `dump_as_json`, else I print the result.